testcontainers/opensearch.py

```diff
@@ -0,0 +1,105 @@
+from opensearchpy import OpenSearch
+from opensearchpy.exceptions import ConnectionError, TransportError
+
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.waiting_utils import wait_container_is_ready
+
+
+class OpenSearchContainer(DockerContainer):
+    """
+    The following example demonstrates how to create a new index in an OpenSearch container
+    and add a document to it. It also shows how to search within the created index. The refresh
+    step in between makes sure that the newly created document is available for search.
+
+    The method :code:`get_client` can be used to create a OpenSearch Python Client.
+    The method :code:`get_config` can be used to retrieve the host, port, user
+    and password of the container.
+
+    Example
+    -------
+    .. doctest::
+
+        >>> from testcontainers.opensearch import OpenSearchContainer
+
+        >>> with OpenSearchContainer() as opensearch:
+        ...   client = opensearch.get_client()
+        ...   creation_result = client.index(index="test", body={"test": "test"})
+        ...   refresh_result = client.indices.refresh(index="test")
+        ...   search_result = client.search(index="test", body={"query": {"match_all": {}}})
+    """
+
+    def __init__(
+        self,
+        image="opensearchproject/opensearch:2.4.0",
+        port_to_expose=9200,
+        security_disabled=True,
+        **kwargs,
+    ):
+        """
+        Args:
+            image (str, optional): The Docker image to use for the container.
+                                   Defaults to "opensearchproject/opensearch:2.4.0".
+            port_to_expose (int, optional): The port to expose on the container.
+                                            Defaults to 9200.
+            security_disabled (bool, optional): `True` disables the security plugin in OpenSearch.
```

Comment:
What do you think about changing this to `security_enabled` to avoid any confusion about double negatives?

Reply:
No strong opinion on this, can live with both, tweaked it.